### ISSUE TITLE
MaterialPipelineKey should have public fields

### ISSUE THREAD

**Issue by Shatur** [bevyengine/bevy#4507]
In 0.6 [MaterialPipeline](https://docs.rs/bevy/0.6.0/bevy/pbr/struct.MaterialPipeline.html) used `(MeshPipelineKey, M::Key)` tuple as a key, where `M` is a custom material. But in 0.7 after https://github.com/bevyengine/bevy/pull/3959 it started using [MaterialPipelineKey](https://docs.rs/bevy/0.7.0/bevy/pbr/struct.MaterialPipelineKey.html) struct which is defined in the following way:

```rust
#[derive(Eq, PartialEq, Clone, Hash)]
pub struct MaterialPipelineKey<T> {
    mesh_key: MeshPipelineKey,
    material_key: T,
}
```

It look nicer, but all fields are private, so I can't call `specialize` in my user code for such materials (because the function accepts a key).

[Here](https://github.com/cryscan/bevy-hikari/blob/dfe7671409f38470237a132c7b5781544a9d95d2/src/overlay.rs#L482) is how a real project uses it for overlay.
And [here](https://github.com/cryscan/bevy-hikari/pull/6) is my PR where I ported it. But it doesn't compile since in [the mentioned place](https://github.com/gardum-game/bevy-hikari/blob/bevy-0.7/src/overlay.rs#L485) above I have to create `MaterialPipelineKey` (it compiles if I comment it out).

**Comment by superdump** [comment]
It looks like the fields were private as `MaterialPipelineKey` only used in `MaterialPlugin`. In the `bevy-hikari` case, it does not use `MaterialPlugin` because `MaterialPlugin` only queues meshes with the specialized materials to the opaque / alpha mask / transparent render phases, and `bevy-hikari` has its own phases to queue to. So it uses the `SpecializedMaterial` abstraction, but not the `MaterialPlugin` and then implements its own queuing. This seems like a shortcoming of the API at the moment. Making the fields public does seem like a quick way to regain the flexibility that was there in 0.6 without requiring significant code duplication or significant design effort to make the material plugin handle specification of the phase(s) to support.

**Comment by Shatur** [comment]
Fixed by #4508.
